File: TobyAndSheeps/solutions/carlos.cc
#include<bits/stdc++.h>
using namespace std;

#define endl '\n'

const int MAXN = 100000;

int main() {
  int n, q, l, r, x;
  while (cin >> n) {
    assert(n <= MAXN);
    vector<int> values(n + 1);
    for (int i = 1; i <= n; ++i) {
      cin >> x;
      assert(x < 2);
      values[i] = values[i - 1] + x;
    }
    cin >> q;
    assert(q <= MAXN);
    for (int i = 0; i < q; ++i) {
      cin >> l >> r;
      assert(l <= n);
      assert(r <= n);
      assert(l <= r);
      int size_interval = r - l + 1;
      int sheeps = values[r] - values[l - 1];
      int wolves = size_interval - sheeps;
      cout << ((wolves > sheeps) ? "Yes" : "No") << endl;
    }
  }
  return 0;
}

File: TobyAndSheeps/solutions/carlos.java
import java.io.*;
import java.util.StringTokenizer;

public class Template {

    public static void main(String []args) throws IOException {
        Scanner in = new Scanner(System.in);
        OutputWriter out = new OutputWriter(System.out);
        Task solver = new Task();
        solver.solve(in, out);
        out.close();
    }

}

class Task{
    public void solve(Scanner in, OutputWriter out)  throws IOException {
        int n = in.nextInt();
        int []values = new int[n + 1];
        int x;
        for (int i = 1; i <= n; i++) {
          x = in.nextInt();
          values[i] = values[i - 1] + x;
        }
        int q = in.nextInt();
        for (int i = 0; i < q; i++) {
          int l = in.nextInt();
          int r = in.nextInt();
          int sheeps = values[r] - values[l - 1];
          int wolves = (r - l + 1) - sheeps;
          out.print((wolves > sheeps ? "Yes" : "No") + "\n");
        }
    }
}

class Scanner{
    public BufferedReader reader;
    public StringTokenizer st;

    public Scanner(InputStream stream){
        reader = new BufferedReader(new InputStreamReader(stream));
        st = null;
    }

    public String next(){
        while(st == null || !st.hasMoreTokens()){
            try{
                String line  = reader.readLine();
                if(line == null) return null;
                st =  new StringTokenizer(line);
            }catch (Exception e){
                throw (new RuntimeException());
            }
        }
        return st.nextToken();
    }

    public int nextInt(){
        return Integer.parseInt(next());
    }
    public long nextLong(){
        return Long.parseLong(next());
    }
    public double nextDouble(){
        return Double.parseDouble(next());
    }
}

class OutputWriter{
    BufferedWriter writer;

    public OutputWriter(OutputStream stream){
        writer = new BufferedWriter(new OutputStreamWriter(stream));
    }

    public void print(int i) throws IOException {
        writer.write(i);
    }

    public void print(String s) throws IOException {
        writer.write(s);
    }

    public void print(char []c) throws IOException {
        writer.write(c);
    }
    public  void close() throws IOException {
        writer.close();
    }

}
